### ISSUE TITLE
Formatting changes semantics of the code when using functions with the expression body written in the next line

### ISSUE THREAD

**Issue by antifriz** [pinterest/ktlint#1064]
## Expected Behavior
main.kt:
```kotlin
fun foo()
= 1
```
should either leave it as it is or indent it accordingly 

## Observed Behavior
running `ktlint -F main.kt` edits the file in the following way 
```kotlin
fun foo()
1
```
which changes the semantics of the code / makes it invalid
## Steps to Reproduce
```bash
printf 'fun foo()\n=1' > /tmp/foo.kt
ktlint -F /tmp/foo.kt
cat /tmp/foo.kt
```

## Your Environment
* Version of ktlint used: 0.40.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): bash-3.2
* Version of Gradle used (if applicable): n/a
* Operating System and version: macOS 10.15.6
* Link to your project (if it's a public repository): n/a

**Comment by Tapchicoma** [comment]
Could not reproduce with current snapshot version. Most probably was fixed with #1039
